### PR TITLE
Arch patches

### PR DESCRIPTION
**Pull request by jpmedley**
This PR reworks how asynchronous methods are documented. Specifically:

* Puts promises first since they're the more modern pattern.
* Adds information about feature detecting promises.
* Removes information about using JavaScript methods because it's out of scope.

### CODE REVIEW

File: site/en/docs/extensions/mv3/architecture-overview/index.md [hunk 1]
```diff
@@ -234,65 +234,33 @@ using [chrome.tabs.create()][api-create-tab] instead.
...
 
-A method is asynchronous when the callback parameter is available in its signature.
+* Use the returned promise.
+* Pass a callback function into the method.
 
-```js
-// Signature for an asynchronous method
-chrome.tabs.query(object queryInfo, function callback)
-```
+Note that these choiced are mutually exclusive. If you pass a callback to a method, no promise
```
Comment (AmySteam) [inline site/en/docs/extensions/mv3/architecture-overview/index.md:244]:
```suggestion
Note that these choices are mutually exclusive. If you pass a callback to a method, no promise
```

File: site/en/docs/extensions/mv3/architecture-overview/index.md [hunk 2]
```diff
@@ -234,65 +234,33 @@ using [chrome.tabs.create()][api-create-tab] instead.
...
+will be returned. If you use the returned promise, do not pass a callback.
 
-If the extension needed to navigate the user's currently selected tab to a new URL, it would need to
-get the current tab's ID and then update that tab's address to the new URL.
+Generally, you should prefer promises to callbacks. Not all methods in extensions APIs support
+promises, but newer methods do. You can verify whether a method supports promises by checking
+its API reference page. If you need to support both promises and callbacks for the same
+function (because your users have older browsers), you can test whether the method returns a
+promise using `typeof` and
+[optional chaning](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Operators/Optional_chaining).
```
Comment (AmySteam) [inline site/en/docs/extensions/mv3/architecture-overview/index.md:252]:
```suggestion
[optional chaining](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Operators/Optional_chaining).
```

File: site/en/docs/extensions/mv3/architecture-overview/index.md (hunk 1 above)
Comment (AmySteam) [reply site/en/docs/extensions/mv3/architecture-overview/index.md:244]: Should this be an aside? 🤔 I think this is important not to miss, but you can make this call.

**Review by AmySteam — CHANGES_REQUESTED** [review]
@jpmedley Great job on the PR 👍 . Just found a couple of details to fix and one observation.

File: site/en/docs/extensions/mv3/architecture-overview/index.md (hunk 1 above)
Comment (jpmedley) [reply site/en/docs/extensions/mv3/architecture-overview/index.md]: I debated that myself. I think it's unlikely that anyone will actually try to use both methods, so I see no reason to draw attention to it.

**Comment by jpmedley** [comment]
PTAL

**Review by AmySteam — APPROVED** [review]

**Pull request merged**